File: src/c/SoundEventData.cpp
/**
*  Harmonia
*
*  Copyright (c) 2018 Hiiragi
*
*  This software is released under the MIT License.
*  http://opensource.org/licenses/mit-license.php
*/

#pragma once

#include "SoundEventData.h"
#include "SoundEventType.h"

SoundEventData::SoundEventData(std::string registeredSoundId, std::string soundId, SoundEventType type)
{
	_rid = registeredSoundId;
	_sid = soundId;
	_type = type;
}

SoundEventData::~SoundEventData()
{

}

std::string SoundEventData::get_rid()
{
	return _rid;
}

std::string SoundEventData::get_sid()
{
	return _sid;
}

SoundEventType SoundEventData::get_type()
{
	return _type;
}

File: src/js/ts/js/src/SoundEventData.ts
class SoundEventData {

    private rid: string;
    private sid: string;
    private type: SoundEventType;

    constructor(registeredSoundId: string, soundId: string, type: SoundEventType) {
        this.rid = registeredSoundId;
        this.sid = soundId;
        this.type = type;
    }

    public getRid(): string {
        return this.rid;
    }

    public getSid(): string {
        return this.sid;
    }

    public getType(): SoundEventType {
        return this.type;
    }

}
